### PR TITLE
log '(empty)' when command stderr is empty

### PR DESCRIPTION
**Pull request by jrm5100**
Sometimes a process will exit with an error code, but without printing anything to stderr.  When this happens, stdout is logged as `(empty)` but stderr is not.

This came up for me because we have a retry strategy that increases memory with each retry.  For an enduser running the workflow, it looks like the error is due to limited memory because there is a `Caused by` entry but not a `Command error` entry.

```
[7c/a9e7a0] Submitted process > Test:TestProcess
[7c/a9e7a0] NOTE: Process `Test:TestProcess` terminated with an error exit status (1) -- Execution is retried (1)
Error executing process > 'Test:TestProcess'

Caused by:
  Process requirement exceed available memory -- req: 20 GB; avail: 16 GB

Command executed:

  exit 1

Command exit status:
  -

Command output:
  (empty)

Work dir:
  /path/to/workdir
```

This change alters the logic of printing `Command error:` to match that of `Command output:`, which I believe makes things slightly more clear.

```
Error executing process > 'Test:TestProcess'

Caused by:
  Process requirement exceed available memory -- req: 20 GB; avail: 16 GB

Command executed:

  exit 1

Command exit status:
  -

Command output:
  (empty)

Command error:
  (empty)

Work dir:
  /path/to/workdir
```

Signed-off-by: John McGuigan <jrm5100@gmail.com>

### CODE REVIEW

File: modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy [hunk 1]
```diff
@@ -1152,15 +1152,17 @@ class TaskProcessor {
...
+            if( lines.size() == 0 ) {
+                message << "  (empty)"
+            }
+            for( String it : lines ) {
+                message << "  ${stripWorkDir(it, task.workDir)}"
             }
+            
             // - this is likely a task wrapper issue
-            else if( task.exitStatus != 0 ) {
+            if( task.exitStatus != 0 & lines.size() == 0) {
```
[the remark this answers is not in the document]
Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy:1165]: Think the bot comment need to be addressed

**Comment by pditommaso** [comment]
The rationale was to only show the error message if there's a message to show. What's the value of showing twice `(empty)` ?

**Comment by jrm5100** [comment]
> The rationale was to only show the error message if there's a message to show. What's the value of showing twice `(empty)` ?

`Caused by` shows the error Nextflow experienced (here due to retrying with more memory until the requirement exceeded what was available) but this obscures the underlying error that caused the retries in the first place.  This was confusing for a user running a workflow because they thought the underlying command needed more memory to run- this was the only part of the message that looked like a cause for the error.

I knew about the retry strategy and that this wasn't the underling issue but I was still tripped up because it looks like any underlying error message may have been dropped (exit status is `-` and no other error message was shown).

In either case I think the current behavior (`Command output` and `Command error` behaving differently) is unexpected and potentially confusing.

**Comment by pditommaso** [comment]
Thanks for opening this pull request, but I think it's preferable to keep in the current form

**Pull request closed without merging**